### ISSUE TITLE
[Issue]: Discrepancy in Log Loss Between XGBClassifier and FLAML's First Iteration

### ISSUE THREAD

**Issue by gcasamat** [microsoft/FLAML#1395]
### Describe the issue

I expected the results from an XGBClassifier to match the first iteration of FLAML when using the same input data and initializing FLAML with the parameters of XGBClassifier. However, the results differ. Specifically, the log loss for the XGBClassifier is 0.0458, while FLAML's first iteration produces a log loss of 0.0532 (see the simulation code and logs below)

Could you please clarify why there is a discrepancy in the log loss? 

Thank you!

### Steps to reproduce

```
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import log_loss
from xgboost import XGBClassifier
from flaml import AutoML

def compare(X, y, xgb_params):
    """Compare XGBoost standard avec la première itération de FLAML"""
    
    # Split des données (identique pour les deux cas)
    X_train, X_val, y_train, y_val = train_test_split(
        X, y, test_size=0.2, random_state=42  # Assurez un split déterministe
    )
    
    # 1. XGBoost standard
    print("Configuration XGBoost initiale:")
    print("-" * 80)
    for k, v in sorted(xgb_params.items()):
        print(f"{k}: {v}")
    
    print("\nEntraînement XGBoost standard...")
    xgb_model = XGBClassifier(**xgb_params)
    xgb_model.fit(
        X_train, 
        y_train,
        eval_set=[(X_val, y_val)],
        verbose=False
    )
    
    xgb_probs = xgb_model.predict_proba(X_val)
    xgb_score = log_loss(y_val, xgb_probs)
    print("\nScore log_loss (XGBoost):", xgb_score)
    
    # 2. FLAML première itération
    print("\nConfiguration et entraînement FLAML...")
    automl = AutoML()

    # Espace de recherche fixe pour FLAML
    custom_hp = {
        'xgboost': {
            param: {'domain': value, 'type': 'fixed'} for param, value in xgb_params.items()
        }
    }
    
    flaml_settings = {
        'time_budget': 1,  # Limité à une seconde
        'metric': 'log_loss',
        'task': 'classification',
        'estimator_list': ['xgboost'],
        'eval_method': 'auto',  # Évaluation explicite sur X_val
        'verbose': 3,
        'custom_hp': custom_hp,
        'log_file_name': 'flaml.log',
        'split_ratio': 0,  # Désactiver tout fractionnement interne
        'X_val': X_val,  # Utiliser explicitement le même ensemble de validation
        'y_val': y_val,
        'retrain_full':False,
        'n_jobs': 1,  # Réduit la variabilité due à l'exécution parallèle
        'early_stop': False,  # Empêche les arrêts précoces
        'sample': False
    }
    
    automl.fit(X_train, y_train, **flaml_settings)
    
    # Résultats de FLAML
    best_model = automl.model.estimator
    flaml_probs = best_model.predict_proba(X_val)
    flaml_score = log_loss(y_val, flaml_probs)
    
    print("\nScore log_loss de FLAML:", flaml_score)
    print("\nComparaison finale:")
    print(f"XGBoost log_loss:   {xgb_score:.6f}")
    print(f"FLAML log_loss:     {flaml_score:.6f}")
    print(f"Différence:         {abs(xgb_score - flaml_score):.6f}")
    
    return {
        'xgb_score': xgb_score,
        'flaml_score': flaml_score,
        'flaml_best_config': automl.best_config
    }

# Configuration des paramètres
xgb_params = {
    'n_estimators': 100,
    'max_depth': 6,
    'learning_rate': 0.3,
    'subsample': 1.0,
    'colsample_bytree': 0.8,
    'colsample_bylevel': 1.0,
    'colsample_bynode': 1.0,
    'min_child_weight': 1,
    'reg_alpha': 0.0,
    'reg_lambda': 1.0,
    'random_state': 42,
    'tree_method': 'hist',
    'objective': 'binary:logistic',
    'eval_metric': 'logloss',
    'booster': 'gbtree'
}

if __name__ == "__main__":
    # Génération de données test
    np.random.seed(42)
    X = np.random.randn(1000, 20)
    y = (X[:, 0] + X[:, 1] > 0).astype(int)
    
    # Exécution de la fonction
    results = compare(X, y, xgb_params)

```

### Screenshots and logs

Configuration XGBoost initiale:
--------------------------------------------------------------------------------
booster: gbtree
colsample_bylevel: 1.0
colsample_bynode: 1.0
colsample_bytree: 0.8
eval_metric: logloss
learning_rate: 0.3
max_depth: 6
min_child_weight: 1
n_estimators: 100
objective: binary:logistic
random_state: 42
reg_alpha: 0.0
reg_lambda: 1.0
subsample: 1.0
tree_method: hist

Entraînement XGBoost standard...

Score log_loss (XGBoost): 0.04581556738621408

Configuration et entraînement FLAML...
[flaml.automl.logger: 01-23 09:39:12] {1728} INFO - task = classification
[flaml.automl.logger: 01-23 09:39:12] {1736} INFO - Data split method: stratified
[flaml.automl.logger: 01-23 09:39:12] {1739} INFO - Evaluation method: holdout
[flaml.automl.logger: 01-23 09:39:12] {1838} INFO - Minimizing error metric: log_loss
[flaml.automl.logger: 01-23 09:39:12] {1955} INFO - List of ML learners in AutoML Run: ['xgboost']
[flaml.automl.logger: 01-23 09:39:12] {2258} INFO - iteration 0, current learner xgboost
[flaml.automl.logger: 01-23 09:39:12] {2393} INFO - Estimated sufficient time budget=163s. Estimated necessary time budget=0s.
[flaml.automl.logger: 01-23 09:39:12] {2442} INFO -  at 0.0s,	estimator xgboost's best error=0.0532,	best estimator xgboost's best error=0.0532
[flaml.automl.logger: 01-23 09:39:12] {2258} INFO - iteration 1, current learner xgboost
[flaml.automl.logger: 01-23 09:39:12] {2442} INFO -  at 0.0s,	estimator xgboost's best error=0.0488,	best estimator xgboost's best error=0.0488
[flaml.automl.logger: 01-23 09:39:12] {2258} INFO - iteration 2, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.1s,	estimator xgboost's best error=0.0488,	best estimator xgboost's best error=0.0488
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 3, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.1s,	estimator xgboost's best error=0.0485,	best estimator xgboost's best error=0.0485
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 4, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.1s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 5, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.1s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 6, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.2s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 7, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.2s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 8, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.2s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 9, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.2s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 10, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.3s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 11, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.3s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 12, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.3s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 13, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.3s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 14, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.4s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 15, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.4s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 16, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.4s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 17, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2442} INFO -  at 0.4s,	estimator xgboost's best error=0.0458,	best estimator xgboost's best error=0.0458
[flaml.automl.logger: 01-23 09:39:13] {2258} INFO - iteration 18, current learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2470} INFO - stop trying learner xgboost
[flaml.automl.logger: 01-23 09:39:13] {2550} INFO - selected model: XGBClassifier(base_score=None, booster='gbtree', callbacks=[],
              colsample_bylevel=1.0, colsample_bynode=1.0, colsample_bytree=0.8,
              device=None, early_stopping_rounds=None, enable_categorical=False,
              eval_metric='logloss', feature_types=None, gamma=None,
              grow_policy=None, importance_type=None,
              interaction_constraints=None, learning_rate=0.3, max_bin=None,
              max_cat_threshold=None, max_cat_to_onehot=None,
              max_delta_step=None, max_depth=6, max_leaves=6,
              min_child_weight=1, missing=nan, monotone_constraints=None,
              multi_strategy=None, n_estimators=100, n_jobs=1,
              num_parallel_tree=None, random_state=42, ...)
[flaml.automl.logger: 01-23 09:39:13] {1985} INFO - fit succeeded
[flaml.automl.logger: 01-23 09:39:13] {1986} INFO - Time taken to find the best model: 0.11479568481445312

Score log_loss de FLAML: 0.04575710963728973

Comparaison finale:
XGBoost log_loss:   0.045816
FLAML log_loss:     0.045757
Différence:         0.000058

**Comment by thinkall** [comment]
Thank you @gcasamat for reporting this. You've missed the `init_value` in the `custom_hp` and `max_leaves` in  the params. With below  changes:
```python
...
    # Espace de recherche fixe pour FLAML
    custom_hp = {
        'xgboost': {
            param: {'domain': value, 'init_value': value, 'type': 'fixed'} for param, value in xgb_params.items()
        }
    }
...
# Configuration des paramètres
xgb_params = {
    'max_leaves': 31,
    'n_estimators': 100,
    'max_depth': 6,
...
```

I get below result:
```
Comparaison finale:
XGBoost log_loss:   0.049057
FLAML log_loss:     0.049057
Différence:         0.000000
```

**Comment by gcasamat** [comment]
It worked, thank you!
